quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java

```diff
@@ -64,15 +78,96 @@ public String apply(String value, ConfigSourceInterceptorContext configSourceInt
                         .paramLabel("format")
+                        .build(),
+                builder().from("log-console-color")
+                        .to("quarkus.log.console.color")
+                        .defaultValue(Boolean.TRUE.toString())
+                        .description("Enable or disable colors when logging to console.")
+                        .paramLabel(Boolean.TRUE + "|" + Boolean.FALSE)
+                        .build(),
+                builder().from("log-console-enabled")
+                        .mapFrom("log")
+                        .to("quarkus.log.console.enable")
+                        .hidden(true)
+                        .transformer(resolveLogHandler(DEFAULT_LOG_HANDLER))
+                        .build(),
+                builder().from("log-file-enabled")
+                        .mapFrom("log")
+                        .to("quarkus.log.file.enable")
+                        .hidden(true)
+                        .transformer(resolveLogHandler("file"))
+                        .build(),
+                builder().from("log-file")
+                        .to("quarkus.log.file.path")
+                        .defaultValue(DEFAULT_LOG_PATH)
+                        .description("Set the log file path and filename.")
+                        .paramLabel("<path>/<file-name>.log")
+                        .transformer(LoggingPropertyMappers::resolveFileLogLocation)
+                        .build(),
+                builder().from("log-file-format")
+                        .to("quarkus.log.file.format")
+                        .defaultValue("%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p [%c] (%t) %s%e%n")
+                        .description("Set a format specific to file log entries.")
+                        .paramLabel("<format>")
                         .build()
         };
     }
 
+    private static BiFunction<String, ConfigSourceInterceptorContext, String> resolveLogHandler(String handler) {
+        return (parentValue, context) -> {
+
+            //we want to fall back to console to not have nothing shown up when wrong values are set.
+            String consoleDependantErrorResult = handler.equals(DEFAULT_LOG_HANDLER) ? Boolean.TRUE.toString() : Boolean.FALSE.toString();
+
+            if(parentValue.isBlank()) {
+                addInitializationException(Messages.emptyValueForKey("log"));
+                return consoleDependantErrorResult;
+            }
+
+            String[] logHandlerValues = parentValue.split(",");
+
+            if (!AVAILABLE_LOG_HANDLERS.containsAll(List.of(logHandlerValues))) {
+                addInitializationException(Messages.notRecognizedValueInList("log", parentValue, String.join(",", AVAILABLE_LOG_HANDLERS)));
+                return consoleDependantErrorResult;
+            }
+
+            for (String handlerInput : logHandlerValues) {
```

Comment:
isn't this condition redundant given the check at line 147?

Reply:
eh.. no? :D sorry dont know what you mean. In lines 140 to 150 i sanity check the input with some guar clauses, as its not always coming from CLI and thus `expectedValues` doesn't add sanity checking before quarkus starts. In line 152ff i actually transform the sanitized value to the underlying true/false for quarkus and return that value so its set internally.

Reply:
got it finally 👍  thanks